src/Agent.Worker/Handlers/LegacyPowerShellHandler.cs

```diff
@@ -181,7 +181,17 @@ public async Task RunAsync()
             ArgUtil.Directory(TaskDirectory, nameof(TaskDirectory));
 
             // Warn about legacy handler.
-            ExecutionContext.Warning($"Task '{this.Task.Name}' ({this.Task.Version}) is using deprecated task execution handler. The task should use the supported task-lib: https://aka.ms/tasklib");
+            if (!ExecutionContext.Variables.Allow_Legacy_Powershell)
+            {
+                ExecutionContext.Result = TaskResult.Failed;
+                ExecutionContext.Error($"Task '{this.Task.Name}' ({this.Task.Version}) is using deprecated task execution handler. The task should use the supported task-lib: https://aka.ms/tasklib");
+                return;
+            }
+            else
+            {
+                ExecutionContext.Warning($"Task '{this.Task.Name}' ({this.Task.Version}) is using deprecated task execution handler. The task should use the supported task-lib: https://aka.ms/tasklib");
```

Comment:
"This task must be updated to use the supported..."

Reply:
And let's pick an agent version where we'll remove it. 2.166?

Reply:
That's March 2020, updated to reflect that - is that late enough for on-prem?